path_point[] point_array
path_point[] obstacle_array